\newcommand{\gf}{gain-of-function allele}
\newcommand{\lf}{loss-of-function allele}
\newcommand{\strong}{strong loss-of-function allele}
\newcommand{\weak}{weak loss-of-function allele}

% gene names
\newcommand{\ras}{\gene{let-60} (\emph{ras})}
\newcommand{\rasp}{\protein{let-60}}
\newcommand{\dpy}[1]{\gene{dpy-22#1}}
\newcommand{\letgfn}{3,021}
\newcommand{\letlfn}{857}
\newcommand{\letgf}{\gene{let-60(gf)}}
\newcommand{\letlf}{\gene{let-60(lf)}}
\newcommand{\strongn}{2,036}
\newcommand{\weakn}{266}
\newcommand{\transn}{2,128}
\newcommand{\bx}{\dpy{(bx93)}}
\newcommand{\sy}{\dpy{(sy622)}}

% more space between rows
\newcommand{\ra}[1]{\renewcommand{\arraystretch}{#1}}

% document begins here
\section*{Abstract}
  \textbf{Although transcriptomes have recently been used as phenotypes with
  which to perform epistasis analyses, they are not yet used to study intragenic
  function/structure relationships. We developed a theoretical framework to
  study allelic series using transcriptomic phenotypes. As a proof-of-concept,
  we apply our methods to an allelic series of \dpy{}, a highly pleiotropic
  \emph{Caenorhabditis~elegans} gene orthologous to the human gene \gene{MED12},
  which encodes a subunit of the Mediator complex. Our methods identify
  functional units within \dpy{} that modulate Mediator activity upon various
  genetic programs, including the Wnt and Ras modules.}

\section*{Introduction}
Mutations of a gene can yield a series of alleles with different phenotypes that
reveal multiple functions encoded by that gene, regardless of the alleles'
molecular nature. In \emph{Caenorhabditis~elegans}, allelic series have
characterized genes such as \gene{let-23/EGFR}, \gene{lin-3/EGF} and
\gene{lin-12/NOTCH}~\citep{Aroian1991, Ferguson1985a, Greenwald1983}. Allelic
series provide a way to probe genes where biochemical approaches would be
difficult, slow or uninformative with regards to the biological phenomenon of
interest. Their power derives from the ability to draw broad conclusions about
the gene of interest in terms of gene dosage and functional units, to the extent
that these two factors are separable, without regard to the molecular identity
of the mutations that created these alleles. Here, gene dosage is defined as the
combined effects of transcriptional and translational expression, gene product
localization, and biochemical kinetics of the final gene product \emph{in situ}.
To study allelic series, we must first enumerate the phenotypes each allele
affects, and subsequently order the alleles into severity and dominance
hierarchies per phenotype. The resulting hierarchies enable us to better
understand how a given gene, which may be highly pleiotropic, can give rise to
highly specific mutant phenotypes when mutated in just the right way.

Biology has moved from expression measurements of single genes towards
genome-wide measurements. Expression profiling via RNA-seq~\citep{Mortazavi2008}
enables simultaneous measurement of transcript levels for all genes in a genome,
yielding a transcriptome. These measurements can be made on whole organisms,
isolated tissues, or single cells~\citep{Tang2009,Schwarz2012}. Transcriptomes
have been successfully used to identify new cell or organismal
states~\citep{Angeles-Albores2017,Villani2017}. Transcriptomic states can be
used to perform epistatic analyses~\citep{Dixit2016,AngelesAlboresHIF},
but have not been used to characterize allelic series.

We have devised methods for characterizing allelic series using RNA-seq. To test
these methods, we selected three alleles~\citep{Zhang2000,Moghal2003} of a
\cel{} Mediator complex subunit gene, \dpy{}. Mediator is a macromolecular
complex with $\sim25$ subunits~\citep{Jeronimo2017} that globally regulates RNA
polymerase II (Pol II)~\citep{Allen2015,Takagi2006}. The Mediator complex has at
least four biochemically distinct modules: the Head, Middle and Tail modules and
a CDK-8-associated Kinase Module (CKM). The CKM associates reversibly with other
modules, and appears to inhibit transcription~\citep{Knuesel2009,Elmlund2006}.
In \cel{} development, the CKM promotes the formation of the male
tail~\citep{Zhang2000} (through interactions with the Wnt pathway), as well as
formation of the hermaphrodite vulva~\citep{Moghal2003} (through inhibition of
the Ras pathway). Null alleles of \dpy{} are likely to be lethal, based on
embryonic lethal phenotypes observed after RNAi~\citep{Wang2004,Lehner2006} and
the severe phenotypes of a strong \dpy{} hypomorphic allele, \gene{dpy-22(e652)}
(homozygous hermaphrodites are very sick)~\citep{Riddle1997}. Homozygotes of
allele \gene{dpy-22(bx93)}, which encodes a premature stop codon
Q2549Amber~\citep{Zhang2000}, appear grossly wild-type, though this allele does
not have complete wild-type functionality, since it fails to fully complement
the Muv phenotype of another allele, \emph{sy622}, in a sensitized \emph{let-23}
background. In contrast, animals homozygous for a more severe allele,
\gene{dpy-22(sy622)} encoding another premature stop codon,
Q1698Amber~\citep{Moghal2003}, are dumpy (Dpy), have egg-laying defects (Egl),
and have multiple vulvae (Muv) (Fig.~\ref{fig:genemodel}). In
humans, \protein{MED12} is known to have a proline-, glutamine- and leucine-rich
domain that interacts with the WNT pathway~\citep{Kim2006}. However, many
disease-causing variants fall outside of this domain~\citep{Yamamoto2015}.
In spite of its causative role in a number of neurodevelopmental
disorders~\citep{Graham2013}, the structural and functional features of this
gene are poorly understood, partially because genetic approaches towards
studying pleiotropic genes have proved difficult in the past, highlighting the
need for new methods.

\begin{figure}
  \centering{}
  \includegraphics[width=0.5\textwidth]{alleleims/Gene_Model.pdf}
  \caption{
           Protein sequence schematic for \protein{dpy-22}. The positions of the
           nonsense mutations used are shown.
           }
\label{fig:genemodel}
\end{figure}


\section*{Methods}\label{sec:methods}

\subsection*{Strains used}
Strains used were N2 wild-type (Bristol)~\citep{Brenner1974},
PS4087 \sy{}~\citep{Moghal2003},
PS4187 \bx{}~\citep{Zhang2000},
PS4176 \gene{dpy-6(e14) dpy-22(bx93)/+ dpy-22(sy622)}~\citep{Moghal2003},
MT4866 \gene{let-60(n2021)}~\citep{Beitel1990a},
MT2124 \gene{let-60(n1046gf)}~\citep{Beitel1990a} and
EW15 \gene{bar-1(ga80)}~\citep{Eisenmann1998}.
Lines were grown on standard nematode growth media (NGM) Petri plates seeded
with OP50 \ecol{} at 20\degree{}C~\citep{Brenner1974}.

\subsection*{Strain synchronization, harvesting and RNA sequencing}
With the exception of strain MT4866, strains were synchronized by bleaching
P$_0$'s into virgin S. basal (no cholesterol or ethanol added) for 16--18 hours.
Arrested L1 larvae were placed in NGM plates seeded with OP50 at 20\degree{}C
and grown to the young adult stage (assessed by vulval morphology and lack of
embryos). We discovered that MT4866 dies upon L1 starvation for this period of
time. As a result, we synchronized this strain by double bleaching. Animals were
picked if they were young adults, regardless of whether any vulval or
morphological phenotypes were present. RNA extraction and sequencing was
performed as previously described by~\citet{AngelesAlboresHIF,
Angeles-Albores2017}. Briefly, young adults were placed in 10~$\mu$L of TE
buffer,  and digested using  Recombinant Proteinase K PCR Grade (Roche Lot 656
No. 03115 838001) incubated with 1\% SDS 657 and 1.25~$\mu$L RNA Secure (Ambion
AM7005). Total RNA was extracted using the Zymo Research Directzol RNA MicroPrep
Kit (Zymo Research, SKU R2061).\@ mRNA was subsequently purified using a NEBNext
Poly(A) mRNA Magnetic Isolation Module (New England Biolabs, NEB, \#E7490).
Sequencing libraries were generated using the NEBNext Ultra RNA Library Prep Kit
for Illumina (NEB \#E7530). These libraries were sequenced using an Illumina
HiSeq2500 machine in single-read mode with a read length of 50 nucleotides.

\subsection*{Read pseudo-alignment and differential expression}
Reads were pseudo-aligned to the \cel{} genome (WBcel235) using
Kallisto~\citep{Bray2016}, using 200 bootstraps and with the sequence bias
(\texttt{--seqBias}) flag. The fragment size for all libraries was set to 200
and the standard deviation to 40. Quality control was performed on a subset of
the reads using FastQC, RNAseQC, BowTie and MultiQC~\citep{Andrews2010,
Deluca2012, Langmead2009, Ewels2016}.

Differential expression analysis was performed using
Sleuth~\citep{Pimentel2016a}. We used a general linear model to identify genes
that were differentially expressed between wild-type and mutant libraries. To
increase our statistical power, we pooled young adult wild-type replicates from
other published~\citep{AngelesAlboresHIF, Angeles-Albores2017} and unpublished
analyses adjusting for batch effects. Briefly, batch effects were controlled by
including the identity of the person who collected the worms and the method
by which the libraries were generated as covariates.å

\subsection*{False hit analysis}
To accurately count phenotypes, we developed a false hit algorithm
(Algorithm~\ref{alg:false}). We implemented this algorithm for comparisons of
three genotypes using Python. Such an experiment can result in $128$ possible
combinations of phenotypic classes (ignoring size). This large number
of models necessitates an algorithmic approach that can restrict the
number of models. Our algorithm uses a noise function that assumes
false hit events are non-overlapping (i.e.\ the same gene cannot be the result
of two false positive events in two or more genotypes) to determine the average
noise flux between phenotypic classes. These assumptions break down if
false-positive or negative rates are large (>25\%).

To benchmark our algorithm, we generated one thousand Venn diagrams at random.
For each Venn diagram, we calculated the average false positive and false
negative flux matrices. Then, we added noise to each phenotypic class in the
Venn diagram, assuming that fluxes were normally distributed with mean and
standard deviation equal to the flux coefficient calculated. We input the noised
Venn diagram into our false hit analysis and collected classification
statistics. For a given signal-to-noise cutoff, $\lambda$, classification
accuracy varied significantly with changes in the total error rate. In the
absence of false negative hits, false hit analysis can accurately identify
non-empty genotype-associated phenotypic classes, but identifying
genotype-specific classes becomes difficult if the experimental false positive
rate is high. On the other hand, even moderate false negative rates ($>10\%$)
rapidly degrade signal from genotype-associated classes. For classes that are
associated with three genotypes, an experimental false negative rate of 30\% is
enough on average to prevents this class from being observed.

We selected $\lambda=3$ because classification using this threshold was high
across a range of false positive and false negative combinations. A challenge to
applying this algorithm to our data is the fact that the false negative rate for
our experiment is unknown. Although there has been significant progress in
controlling and estimating false positive rates, we know of no such attempts for
false negative rates. It is unlikely that the false negative rate for our study
is lower than the false positive rate, because all genotypes except the controls
are likely underpowered. We used false negative rates between 10--20\% for false
hit analysis. All analyses returned the same final model.

We asked whether re-classification of some classes into others could improve
model fit. We manually re-classified the (\sy{},\bx{})-associated and the (\bx,
\emph{trans-heterozygote})-associated classes into the \emph{bx93}-associated
class (which is associated with all genotypes), and compared $\chi^2$ statistics
between a re-classified reduced model ($\chi^2=72$) and a reduced model
($\chi^2=130$). Based on the lower $\chi^2$ of the re-classified reduced model,
we concluded that it is the most likely model given our data.

\begin{algorithm}[H]
  \DontPrintSemicolon{}
  \KwData{$\mathbf{M}_{obs} =  \{N_l\}$, an observed set of classes, where each
  class is labelled by $l\in L$ and is of size $N_l$. $f_p, f_n$, the false
  positive and negative rates respectively. $\alpha$, the signal-to-noise
  threshold for acceptance of a class.}
  \KwResult{$\mathbf{M}_{reduced}$, a reduced model that fits the data.}
  \BlankLine{}
  \Begin{
    \emph{Define a minimal model}, $\mathbf{K}$\;

    \emph{Refine the model until convergence or iterations max out}\; \\
    $i \leftarrow 0$\; \\
    $\mathbf{K_{prev}} \leftarrow \emptyset$\; \\
    \While{$(i < i_{\max})~|~(\mathbf{K_{prev}} \neq \mathbf{K}$)}{
      $\mathbf{K_{prev}} \leftarrow \mathbf{K}$\;

      \emph{Define a noise function to estimate error flows in
            $\mathbf{K}$}\;
      $\mathbf{F} \leftarrow \textrm{noise}(\mathbf{K}, f_p, f_n)$\;

      \For{$l \in L$}{
          \emph{Calculate signal to noise for each labelled class}\;
          \emph{False negatives can result in $\lambda < 0$}\;
          $\lambda_l \leftarrow \mathbf{M}_{obs, l}/F_{l}$\;
          % \emph{Use classes with high $\lambda_l$ to refine the model}\;
              \If{$(\lambda > \alpha)~|~(\lambda < 0)$}{
                $\mathbf{K}_l \leftarrow \mathbf{M}_{obs, l}$\;
              } %if
          } % end for
        $i++$
    } % end while
  } % end begin

  $\mathbf{M}_{reduced} = \mathbf{K}$\;

  \Return{$\mathbf{M}_{reduced}$}\;
  \BlankLine{}\;
  \caption{False Hit Algorithm. Briefly, the algorithm initializes a reduced
           model with the phenotypic class or classes labelled by the largest
           number of genotypes. This reduced model is used to estimate noise
           fluxes, which in turn can be used to estimate a signal-to-noise metric
           between observed and modelled classes. Classes that exhibit a high
           signal-to-noise are incorporated into the reduced model.}
  \label{alg:false}
\end{algorithm}


\subsection*{Dominance analysis}
\label{subsec:dominance}
We modeled allelic dominance as a weighted average of allelic activity:
\begin{equation}
  \beta_{a/b,i,\text{Pred}}(d_a) = d_a\cdot \beta_{a/a,i} +
                                   (1-d_a)\cdot \beta_{b/b,i},
\end{equation}
where $\beta_{k/k, i}$ refers to the $\beta$ value of the $i$th isoform in a
genotype $k/k$, and $d_a$ is the dominance coefficient for allele $a$.

To find the parameters $d_a$ that maximized the probability of observing the
data, we found the parameter, $d_a$, that maximized the equation:
\begin{equation}
    P(d_a|D,H,I) \propto \prod_{i \in S}
                   \exp{-\frac{{(\beta_{a/b,i,\text{Obs}} -
                                \beta_{a/b,i,\text{Pred}}(d_a))}^2}{
                                2\sigma_i^2}}
\end{equation}
where $\beta_{a/b,i,\text{Obs}}$ was the coefficient associated with the $i$th
isoform in the \emph{trans}-het $a/b$ and $\sigma_i$ was the standard error of
the $i$th isoform in the \emph{trans}-heterozygote samples as output by
Kallisto. $S$ is the set of isoforms that participate in the regression (see
main text). This equation describes a linear regression which was solved
numerically.

\subsection*{Code}
Code was written in Jupyter notebooks~\citep{Perez2007} using the Python
programming language. The Numpy, pandas and scipy libraries were used for
computation~\citep{VanDerWalt2011,McKinney2011,Oliphant2007} and the matplotlib
and seaborn libraries were used for data visualization~\citep{Hunter2007,
Waskom}. Enrichment analyses were performed using the WormBase Enrichment
Suite~\citep{Angeles-Albores2016,Angeles-Albores2018}. For all enrichment
analyses, a $q$-value of less than $10^{-3}$ was considered statistically
significant. For gene ontology enrichment analysis, terms were considered
statistically significant only if they also showed an enrichment fold-change
greater than 2.

\subsection*{Data Availability}
Raw and processed reads were deposited in the Gene Expression Omnibus. Scripts
for the entire analysis can be found with version control in our Github
repository, \url{https://github.com/WormLabCaltech/med-cafe}. A user-friendly,
commented website containing the complete analyses can be found at\\ % formatting
\url{https://wormlabcaltech.github.io/med-cafe/}. Raw reads and quantified
abundances for each sample were deposited at the NCBI Gene Expression Omnibus
(GEO)~\citep{Edgar2002} under the accession code GSE107523
(\url{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107523}).

\section*{Results}
\subsection*{RNA-sequencing of three \gene{dpy-22} alleles and two known
             interactor genes}
We carried out RNA-seq on biological triplicates of mRNA extracted from \sy{}
homozygotes, \bx{} homozygotes, and wild type controls, along with
quadruplicates from \emph{trans}-heterozygotes of both alleles with the genotype\\
\gene{dpy-6(e14) dpy-22(bx93)/+ dpy-22(sy622)}. We also sequenced mRNA extracted
from \gene{bar-1(ga80)} (the $\beta$-catenin ortholog in \cel{}),
\gene{let-60(n2021)} and \gene{let-60(n1046gf)} (the Ras ortholog in \cel{})
mutants in triplicate because these genes have been previously described to
interact with \dpy{} to form the vulva~\citep{Moghal2003} and the male
tail~\citep{Zhang2000}. Sequencing was performed at a depth of 20 million reads
per sample. Reads were pseudoaligned using Kallisto~\citep{Bray2016}. We
performed a differential expression using a general linear model specified using
Sleuth~\citep{Pimentel2016a} (see~\hyperref[sec:methods]{Methods}). Differential
expression with respect to the wild type control for each transcript $i$ in a
genotype $g$ is measured via a coefficient $\beta_{g, i}$, which can be loosely
interpreted as the natural logarithm of the fold-change. Transcripts were
considered to have differential expression between wild-type and a mutant if
their false discovery rate, $q$, was less than or equal to 10\%. We used this
method to identify the differentially expressed genes associated with each
mutant (Table~\ref{tab:numbers};
\href{https://wormlabcaltech.github.io/med-cafe/notebook/basic.html}{Basic
Statistics Notebook}) Supplementary File 1 contains all the beta values
associated with this project. We have also generated a website containing
complete details of all the analyses available at the following URL:\@
\url{https://wormlabcaltech.github.io/med-cafe/analysis}.

\begin{table*}
 \centering
 \begin{tabular}{lc}
   \toprule
   Genotype & Differentially Expressed Genes\\
   \midrule
   \bx{} & \weakn{}\\
   \gene{dpy-6(e14)} \dpy{(bx93)} / \emph{+} \dpy{(sy622)} & \transn{}\\
   \sy{} & \strongn{}\\
   \gene{bar-1(ga80)} & 4613\\
   \gene{let-60(n2021)} & 509\\
   \gene{let-60(n1046gf)} & 2526\\
   \bottomrule
 \end{tabular}
 \caption{
          The number of differentially expressed genes relative to the wild-type
          control for each genotype with a significance threshold of 0.1.
          }
\label{tab:numbers}
\end{table*}


\begin{figure}
  \centering{}
  \includegraphics[width=0.45\textwidth]{alleleims/pca.pdf}
  \caption{
           Principal component analysis of the analyzed genotypes. The
           analysis was performed using only those transcripts that were
           differentially expressed in at least one genotype. The plot shows
           that the \emph{trans}-heterozygotes phenocopy the \bx{} homozygotes
           along the first two principal dimensions.
           }
\label{fig:allelic_pca}
\end{figure}

\subsection*{Principal component analysis visualizes the allelic dominance of
             the \bx{} allele over \sy{}}
As a first step in our analysis, we performed dimensionality reduction on the
transcriptomes we sequenced using Principal Component Analysis (PCA). Briefly,
PCA identifies the vectors along which there is most variation in the data.
These vectors can be used to project the data into lower dimensions to assess
whether samples cluster, though interpreting the biological reasons for this
clustering can be challenging. To perform PCA, we selected only those
transcripts that were differentially expressed in at least one genotype, and
used the $\beta$ coefficients associated with these genes to perform PCA.\@
Projecting the data into two dimensions maintains 65\% of the variation. The
first dimension separates the gain and loss of function \gene{let-60} mutants.
The second dimension separates the \dpy{} mutants (Fig.~\ref{fig:allelic_pca}). On the
PCA plot, the \emph{trans}-heterozygote mutants appear to phenocopy the \bx{}
mutants, recapitulating previous experiments that showed the \bx{} allele to be
dominant over the \sy{} allele.

\subsection*{Three \dpy{} genotypes have shared transcriptomic
             phenotypes}
We would like to understand the degree and nature of the dominance
between these \dpy{} alleles. To construct a severity and dominance hierarchy,
we must establish how many transcriptomic phenotypes are represented among the
three \dpy{} genotypes, and of those phenotypes, how many of them are shared
transcriptomic phenotypes (STPs). Shared transcriptomic phenotypes are defined
as the set of genes that are commonly differentially expressed in two mutant
genotypes relative to a wild-type control, regardless of the direction of
change, as defined previously~\citep{AngelesAlboresHIF}. We use the term in the
plural version, because the shared genes may represent multiple independent
modules that formally constitute different phenotypic classes.

We identified significant pairwise STPs between all \dpy{} mutants. The
transcripts that were differentially expressed in \bx{} homozygotes were almost
all differentially expressed in \sy{} homozygotes (189/\weakn{}) and in
\emph{trans}-heterozygotes (192/\weakn{}). On the other hand, although \sy{}
homozygotes and \emph{trans}-heterozygotes exhibited a similar number of
differentially expressed genes, less than half of these were shared between the
two genotypes.

\subsection*{False hit analysis identifies four non-overlapping phenotypic
             classes}

\begin{figure*}
 \centering{}
 \includegraphics[width=\textwidth]{alleleims/flowchart.pdf}
 \caption{
         Flowchart for an analysis of arbitrary allelic series. A set of alleles
         is selected, and the corresponding genotypes are sequenced. Independent
         phenotypic classes are identified, and classes that are the result of
         noise are discarded via a false hit analysis. For each phenotypic
         class, the alleles are ordered in a dominance/complementation
         hierarchy, which can then be used to infer functional regions (FR)
         within the genes in question.
         }
\label{fig:flowchart}
\end{figure*}

Severity and dominance hierarchies must be calculated with respect to each
independent phenotype associated with the alleles under study. A challenge with
expression profiles is to identify these independent phenotypes. We reasoned
that comparing the expression profiles of the two \dpy{} homozygotes and the
\emph{trans}-heterozygote would naturally partition the expression profiles into
groups that would constitute phenotypic classes. However, a three-way comparison
can give rise to 7 ($2^3-1$) possible groupings: transcripts perturbed in only a
single genotype (3), transcripts perturbed in two genotypes (3) and transcripts
perturbed in all three genotypes (1). A shortcoming of RNA-seq is that it is
prone to false positive and false negative artifacts, and these artifacts could
be numerous enough to cause the appearance of certain groups that would not
be there otherwise. In other words, we might find a subset of genes that are
differentially expressed in a single genotype, but if this subset is small
enough, we ought to be concerned that this subset is caused by false positive
hits within this genotype or false negative hits in the other genotypes. This
thought experiment highlights the need to assess which groups have sufficient
statistical support to consider as phenotypic classes.

We developed a method to assess whether groups in a Venn diagram are likely to
be the result of statistical artifacts. Briefly, the algorithm works by first
assuming all of the data is the result of false positive and false negative hits
except for the group of transcripts that is differentially expressed in most
genotypes. Then, using estimates for the false positive and negative response,
we calculate the expected sizes of all the groups after adding noise under this
model. If an observed group is much larger than expected by noise, we refine the
data model to accept the group. This process is iterated until the data model
converges. We called this method a false hit analysis.

We used false hit analysis to identify four non-overlapping phenotypic classes
(Fig.~\ref{fig:flowchart}). We use the term genotype-specific to refer to groups
of transcripts that were perturbed in one mutant genotype. We use the term
genotype-associated to refer to those groups of transcripts whose expression was
significantly altered in two or more mutants genotypes with respect to the wild
type control. The \textbf{\sy{}-associated} phenotypic class consisted of 665
genes differentially expressed in \sy{} homozygotes and in
\emph{trans}-heterozygotes, but which had wild-type expression in \bx{}
homozygotes. The \textbf{\bx{}-associated} phenotypic class contains 229 genes
differentially expressed in all genotypes. The \bx{}-associated class included
re-classified transcripts that had been found to be differentially expressed in
the \bx{} homozygote and one other genotype, because these were very likely to
be the result of false negative hits in the missing genotype, and re-classifying
these transcripts improved our model substantially. We also identified a
\textbf{\sy{}-specific} phenotypic class (1,213 genes) and a
\textbf{\emph{trans}-heterozygote-specific} phenotypic class (1,302 genes; see
the
\href{https://wormlabcaltech.github.io/med-cafe/notebook/phenotypic_classes.html}{
Phenotypic Classes Notebook}).

\subsection*{Severity hierarchy of a \gene{dpy-22} allelic series}
Having separated the expression profiles into phenotypic classes, we can ask
what the severity hierarchy is between the \bx{} allele and the \sy{} allele.
Broadly speaking, there are two ways to assess severity. First, we can ask which
allele causes more mutant phenotypes or phenotypic groups as a homozygote
(\textbf{allelic pleiotropy}). Alternatively, we can identify the allele which
causes the greatest change in expression in a homozygote at each shared
phenotype among the homozygotes of both alleles, which we refer to as
\textbf{allelic magnitude}. An important caveat is that magnitude only makes
sense if the homozygotes of each allele are well correlated (i.e., they have a
linear relationship with small spread). If the phenotypes have zero or negative
correlation between two homozygotes, then the two alleles under inspection are
not of the same kind, i.e., they cannot both be loss-of-function alleles or
gain-of-function alleles for this phenotype, though the converse is not
necessarily true.

The \sy{} homozygote shows more differentially expressed genes that participate
in a greater number of phenotypic classes relative to the \bx{} homozygote.
Thus, the \sy{} allele is a more pleiotropic mutation than the \bx{} allele.
Since the homozygotes of each allele only share a single phenotypic class in
common, we need only assess magnitude along this single phenotype. To calculate
a magnitude coefficient, for genes in the \bx{}-associated phenotypic class, we
plotted the $\beta$ coefficients from the \sy{} homozygote against the $\beta$
coefficients from the \bx{} homozygote (see Fig.~\ref{fig:stp}) and performed a
linear regression to find the slope of this line. Using this method, we found
that the \bx{} homozygote has a magnitude that is 62\% $\pm2\%$ of the \sy{}
homozygote. Taken together, these results suggest that the \sy{} allele
represents a more severe alteration-of-function mutation than the mutation
within the \bx{} allele.

\begin{figure}
  \includegraphics[width=0.5\textwidth]{alleleims/dpy22-stps.pdf}
  \caption{
           Shared Transcriptomic Phenotypes amongst the \dpy{} genotypes are
           regulated in the same direction. For each pairwise comparison, we
           found those transcripts that were commonly differentially expressed
           in both genotypes relative to the wild-type control and plotted the
           $\beta$ coefficients for each. We performed a linear regression on
           each plot to find the line of best fit (broken blue line). Only the
           comparison between \sy{} and \bx{} homozygotes was used to establish
           that the magnitude of the \sy{} allele is greater than the magnitude
           of the \bx{} allele. The other comparisons are shown for
           completeness.
          }
\label{fig:stp}
\end{figure}

\subsection*{Dominance hierarchy of a \gene{dpy-22} allelic series}
We measured allelic dominance for each class using a dominance coefficient
(see~\hyperref[sec:methods]{Methods}). The dominance coefficient is a measure of
the contribution of each allele to the total expression level in
\emph{trans}-heterozygotes. By definition, the \sy{} allele is completely
recessive to \bx{} for the \sy{}-specific phenotypic class. To determine the
dominance coefficient for the remaining phenotypic classes, we first selected
the transcripts within those classes, and asked what linear combination of the
homozygotic $\beta$ coefficients best approximated the $\beta$ coefficients of
the \emph{trans}-heterozygote, subject to the constraint that the sum of the
weights for the two homozygotes should be equal to unity. We solved this problem
by finding the maximum likelihood estimate for these weights. Using this method,
we found that the \sy{} and \bx{} alleles are semidominant ($d_{bx93} = 0.48$)
to each other for the \sy{}-associated phenotypic class. The \bx{} allele is
largely  dominant over the \sy{} allele ($d_{bx93}=0.82$; see
Table~\ref{tab:dom}) for the \bx{}-associated phenotypic class.

\begin{table}
  \centering
  \begin{tabular}{lc}
    \toprule
    Phenotypic Class & Dominance\\
    \midrule
    \sy{}-specific & $1.00\pm0.00$\\
    \sy{}-associated & $0.48\pm0.01$\\
    \bx{}-associated & $0.82\pm0.01$\\
    \bottomrule
    % \midrule{}
  \end{tabular}
  \caption{
           Dominance analysis for the \dpy{/MDT12} allelic series. Dominance
           values closer to 1 indicate \bx{} is dominant over \sy{}, whereas 0
           indicates \sy{} is dominant over \bx{}.
          }
\label{tab:dom}
\end{table}

\subsection*{Phenotypic classes reflect morphological phenotypes}
We performed enrichment analysis of anatomical, phenotypic and gene ontology
terms using the WormBase Enrichment Suite~\citep{Angeles-Albores2016,
Angeles-Albores2018}. The \bx{}-associated phenotypic class was enriched in
genes involved in `immune system processes' (\qval{5}), and was enriched in
genes expressed in the `intestine' (\qval{4}). The \sy{}-associated
class was enriched in genes expressed in the `cephalic sheath
cell' (\qval{4}). Using ontology enrichment analysis from the WormBase
Enrichment Suite, we found that the \sy{}-associated class is enriched in
histones and histone-like proteins (`DNA packaging complex' \qval{3}) as well as
genes involed in `immune system processes' (\qval{5}). The \sy{}-specific class
was enriched in genes that have expression in the `intestine' (\qval{7}),
`muscular system' (\qval{3}) and `epithelial system' (\qval{2}). The genes in
this class are known to cause bacterial lawn avoidance when knocked down or
knocked out (\qval{2}). Finally, GO enrichment showed that the \sy{}-specific
class is specifically enriched in `structural constituents of cuticle'
(\qval{12}), and in genes involved in respiration (\qval{6}). This last result
recapitulates the fact that \sy{} homozygotes show a severe Dumpy phenotype. The
\emph{trans}-heterozygote specific class was enriched in genes expressed in
`male' animals (\qval{63}) and genes expressed in the `reproductive system'
(\qval{21}). GO enrichment of genes in the \emph{trans}-heterozygote specific
class showed enrichment of the genes involved in the `regulation of cell shape'
(\qval{6}) and in a variety of terms involving phosphate metabolism, such as
`nucleoside phosphate binding' (\qval{5}), `dephosphorylation' (\qval{3}) or
`phosphorylation' (\qval{2}), suggesting that this class may be enriched in
genes involved in signal transduction though the reason for this enrichment
remains unclear. The \bx{}-specific class did not show enrichment on any test,
consistent with our interpretation that this class is the result of random false
positive hits.

\subsection*{Predicted interactions of Mediator with Wnt and Ras pathways in
          \cel{}}
Previous work in \cel{}~\citep{Moghal2003,Zhang2000} has implicated \dpy{} as an
inhibitor of the Wnt and Ras pathways during the formation of the vulva and the
male tail. We obtained expression profiles for \gene{bar-1(ga80)} mutants as
well as loss-of-function and gain-of-function Ras mutants, \gene{let-60(n2021)}
and \gene{let-60(n1046gf)} respectively. We predicted that the \sy{}-specific
phenotypic class would exhibit the most significant overlap (assessed by a
hypergeometric enrichment test) with differentially expressed genes in
\gene{let-60(n1046gf)} mutants, whereas the \bx{}-associated phenotypic class
would exhibit the most significant overlap with \gene{bar-1(ga80)} mutants.

The \bx{}-specific class did not show a transcriptomic signature associated with
either the Wnt or the Ras pathway, consistent with our interpretation of this
class as false positive (Fig.~\ref{fig:wnt_stps}). All other classes showed
significant enrichment with genes perturbed in \gene{bar-1(ga80)}. Similarly,
\gene{let-60(n2021)} showed enrichment in all real phenotypic classes, with the
exception of the \emph{trans}-heterozygote specific class. Contrary to our
hypotheses, differentially expressed genes in \gene{let-60(n1046gf)} did not
show significant overlap with the \sy{}-specific phenotype, but they did show
significant overlap with all remaining real phenotypic classes.

\begin{figure}
  \includegraphics[width=0.5\textwidth]{alleleims/stp_pvals.pdf}
  \caption{
          \dpy{} phenotypic classes are statistically significantly enriched
          for signatures of \gene{let-60} (ras) and \gene{bar-1} (wnt)
          signaling.
          We tested whether the overlap between the differentially expressed
          genes in \gene{bar-1(ga80)}, \gene{let-60(n1046gf)} or
          \gene{let-60(n2021)} and the \dpy{} phenotypic classes was
          statistically significant using a hypergeometric enrichment test.
          Since the hypergeometric enrichment test is very sensitive to
          deviations from random, and since we suspect that there may be a broad
          genotoxic response to all mutants, we used a statistical significance
          threshold of $p < 10^{-10}$ (dashed black line).
  }
\label{fig:wnt_stps}
\end{figure}

\section*{Discussion}
\label{sec:conclusions}

\subsection{A conceptual framework for analyses of allelic series using
transcriptomic phenotypes}
Although transcriptomic phenotypes have been used for epistatic
analyses~\citep{Dixit2016,AngelesAlboresHIF,Angeles-Albores2017}, they have not
been used to study gene function in the context of an allelic series.
Outstanding challenges for transcriptomes in allelic series were how to count or
identify distinct phenotypes within the different transcriptomes, how to order
alleles in a severity hierarchy and how to order alleles in a dominance
hierarchy. In this work, we present solutions to these problems, and propose a
set of unifying concepts that we believe will be useful for future analyses. We
re-analyzed an allelic series of the Mediator subunit gene \dpy{} that had been
studied previously~\citep{Moghal2003}, recapitulating and extending previous
results as a proof of principle for our methodology. In our results, we derived
a set of methods that do not rely on the nature of the mutations. In the
subsequent discussion, we use the fact that the mutations we used were
truncations to derive further insights into the functional units present in this
gene.

To interpret our phenotypic classes in a biological context, we investigated
whether these phenotypic classes contained Ras and Wnt expression signatures.
Our attempts were partially successful, but a more rigorous analysis awaits the
availability of a larger mutant set to establish empirically the overlap that is
biologically significant. In part, we reason that some genes may form part of a
broad stress response. If that were the case, many mutants may share similar
transcriptomic signatures.

\subsection*{Phenotypic classes and their sequence requirements}
Because the mutations we used are truncations, our results suggest the existence
of various functional regions in \dpy{/MDT12} (Fig.~\ref{fig:domains}). These
functional regions could encode protein domains with biochemical activity, or
they could encode biochemically active amino acid motifs, such as nuclear
localization sequences or protein binding sites. These functional regions could
confer stability to the protein, thereby regulating its levels. As a caveat, we
note that we have interpreted the effects these mutations have in terms of their
putative effects at the protein level. In the case of our alleles, the relevant
homozygotes had wild-type \dpy{} mRNA levels, suggesting that these mutations
do not affect the stability of the mRNA.

The \sy{}-specific phenotypic class is likely controlled by a single functional
region, functional region 1 (FR1). Sequence necessary for wild-type FR1
functionality is encoded between amino acid positions 1 and 2,549, since this is
the sequence that is intact in the \emph{bx93} allele. We speculate that this
functional region may be the reason that \emph{bx93} is unable to complement the
Muv phenotype of \emph{sy622} in a sensitized \emph{let-23} background, since
\emph{trans}-heterozygotes in this background exhibit a semidominant Muv
phenotype.  The \sy{}-associated phenotypic class is likely controlled by a
second functional region, functional region 2 (FR2), and some necessary
sequences for wild-type function are encoded between amino acid positions 1,698
and 2,549, but additional sequence could lie between amino acids 1 and 1,698. It
is unlikely that FR1 and FR2 are identical because their dominance behaviors are
very different. The \bx{} allele was largely dominant over the \sy{} allele for
the \bx{}-associated class, but gene expression in this class was perturbed in
both homozygotes. The perturbations were greater for \sy{} homozygotes than for
\bx{} homozygotes. This behavior can be explained if the \bx{}-associated class
is controlled jointly by two distinct effectors, functional regions 3 and 4
(FR3, FR4, see Fig.~\ref{fig:domains}). Such a model would propose that the
sequences necessary for FR3 functionality are within the interval 1 and 2,549,
and some sequences necessary for FR4 functionality are encoded between positions
2549 and 3499. This model explains how expression levels of the
\emph{bx93}-associated phenotypic class in the \emph{trans}-heterozygote are
complemented to the levels of the \emph{bx93} homozygote, because FR3 is
complemented in \emph{trans}, but FR4 is defective. Thus, FR3 encodes a
functionality that is not dosage-dependent. One possibility is that FR3 is
equivalent to FR1 or FR2, and FR4 modifies activity of either of these regions
at a subset of loci. A rigorous examination of this model will require studying
many alleles that mutate the region between Q1689 and Q2549 using homozygotes
and \emph{trans}-heterozygotes.

\begin{figure}
  \centering{}
  \includegraphics[width=0.5\textwidth]{alleleims/inferred_domains.pdf}
  \caption{
          The functional regions associated with each phenotypic class can be
          mapped intragenically. The number of genes associated with each class
          is shown. The \bx{}-associated class may be controlled by two
          functional regions. FR1 is a dosage-sensitive unit. FR2 and FR3 could
          be redundant if FR4 is a modifier of FR2 functionality at
          \bx{}-associated loci. Note that the \bx{}-associated phenotypic class
          is actually three classes merged together. Two of these classes are DE
          in \bx{} homozygotes and one other genotype. Our analyses suggested
          that these two classes are likely the result of false negative hits
          and genes in these classes should be differentially expressed in all
          three genotypes, so we merged these three classes together
          (see~\hyperref[sec:methods]{Methods}). }
\label{fig:domains}
\end{figure}

We also found a class of transcripts that had perturbed levels in
\emph{trans}-heterozygotes only; its biological significance is unclear.
Phenotypes unique to \emph{trans}-heterozygotes are often the result of physical
interactions such as homodimerization, or dosage reduction of a toxic
product~\citep{Yook2005}. In the case of \dpy{/MDT12} orthologs, these
explanations seem unlikely since \protein{dpy-22} is a monomeric subunit of the
CKM.\@ Another possibility is that the \emph{trans}-heterozygote-specific class
is the result of complex tissue cross-talk. Massive single-cell RNA-seq of
\cel{} has recently been reported~\citep{Cao2017}, and this tool could provide
valuable information regarding this hypothesis. Another possibility is that the
\emph{cis}-marker we used for the \emph{bx93} allele, \gene{dpy-6(e14)}, which
we assumed to be recessive in all phentoypes, actually has dominant
transcriptomic phenotype.

\subsection*{Occam's razor}
Transcriptomic phenotypes generate large amounts of differential gene expression
data, so false positive and false negative rates can lead to spurious phenotypic
classes whose putative biological significance is misleading. Such
artifacts are particularly likely when a phenotypic class is small. Notably,
errors of interpretation cannot be avoided by setting a more stringent $q$-value
cut-off: doing so will decrease the false positive rate, but increase the false
negative rate, which will in turn produce smaller phenotypic classes than
expected. Our method tries to avoid this pitfall by using total error rate
estimates to assess the plausibility of each class, though a major drawback is
that it relies on a subjective estimation of the false negative rate. These
conclusions are of broad significance to research where highly multiplexed
measurements are compared to identify similarities and differences in the
genome-wide behavior of a single variable under multiple conditions.

We have shown that transcriptomes can be used to study allelic series in the
context of a large, pleiotropic gene. We identified separable phenotypic classes
that would otherwise be obscured by other methods, correlated each class to a
functional region, and identified sequence requirements for each region. Given
the importance of allelic series for characterizing gene function and their
roles in specific genetic pathways, we are optimistic that this method will be a
useful addition to the geneticist's arsenal.
